app/src/main/kotlin/com/wire/android/ui/calling/outgoing/OutgoingCallViewModel.kt

```diff
@@ -96,33 +100,38 @@ class InitiatingCallViewModel @AssistedInject constructor(
 
     private fun onCallEstablished() {
         callRinger.ring(R.raw.ready_to_talk, isLooping = false, isIncomingCall = false)
-        state = state.copy(flowState = InitiatingCallState.FlowState.CallEstablished)
+        state = state.copy(flowState = OutgoingCallState.FlowState.CallEstablished)
     }
 
-    internal suspend fun initiateCall() {
-        val result = startCall(
-            conversationId = conversationId
-        )
-        when (result) {
-            StartCallUseCase.Result.Success -> callRinger.ring(
-                resource = R.raw.ringing_from_me,
-                isIncomingCall = false
-            )
+    @VisibleForTesting
+    suspend fun initiateCall() {
+        observeOutgoingCall().first().run {
+            if (isEmpty()) {
+                val result = startCall(
+                    conversationId = conversationId
+                )
+                when (result) {
+                    StartCallUseCase.Result.Success -> callRinger.ring(
+                        resource = R.raw.ringing_from_me,
+                        isIncomingCall = false
+                    )
 
-            StartCallUseCase.Result.SyncFailure -> {} // TODO: handle case where start call fails
+                    StartCallUseCase.Result.SyncFailure -> {} // TODO: handle case where start call fails
```

Comment:
Should we log error here?

Reply:
Or something else, maybe? I guess the screen will be dismissed without any explaining and the user won't know what happened.

Reply:
This may need a product decision on how to display this error message, let's add some logs here for now